### PR TITLE
Intel compiler as a compiler subsetting

### LINKED ISSUE #5590: Add intel compiler to default settings.yml
Suggested:

```
intel:
        version: ["17.0.0", "17.0.1", "17.0.2", "17.0.3", "17.0.4", "17.0.5", "17.0.6", "17.0.7", "17.0.8",
                  "18.0.0", "18.0.1", "18.0.2", "18.0.3", "18.0.4",
                  "19.0.0", "19.0.1", "19.0.2", "19.0.3"]
        libcxx: [libstdc++, libstdc++11] # Linux only
        runtime: [MD, MT, MTd, MDd] # Windows only
        cppstd: [None, 98, gnu98, 11, gnu11, 14, gnu14, 17, gnu17, 20, gnu20]
```

To investigate:
- Binary compatibility of update/patches: reported erorrs on 17.0.4->17.0.6
- Intel compiler also includes fortran, check if it makes sense regarding cppstd/libcxx
- libcxx/runtime in different OS, how to manage?

### PR DESCRIPTION
**Pull request by danimtb**
Changelog: Add intel compiler support in *settings.yml* and handle compatibility
Docs: missing

- [x] Refer to the issue that supports this Pull Request: closes #5590
- [x] If the issue has missing info, explain the purpose/use case/pain/need that covers this Pull Request.
- [x] I've read the [Contributing guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [x] I've followed the PEP8 style guides for Python code.
- [ ] I've opened another PR in the Conan docs repo to the ``develop`` branch, documenting this one. 

Related to #5626 

<sup>**Note:** By default this PR will skip the slower tests and will use a limited set of python versions. Check [here](https://github.com/conan-io/conan/blob/develop/.github/PR_INCREASE_TESTING.md) how to increase the testing level by writing some tags in the current PR body text.</sup>

-------------------------

Handle Intel compiler as a subsetting of the supported compilers. This also include a first level intel compiler setting to be compatible with the proposal explained in https://github.com/conan-io/conan/issues/5590#issuecomment-519172251

This PRs proposes a way to generate pure intel packages, using directly the `compiler=intel` approach or the `compiler.intel=incompatible` one.

Compatible but different packages can be created for visual studio and gcc using `compiler.intel=compatible`. Moreover, this compatible flag can be specified as input without the intel compiler version (None) in orther to consume intel compiled packages compatible with gcc or visual (`conan install . --profile visual -s compiler.intel=compatible`).

Please review the test cases and we can discuss if this would be the expected behavior.

### CODE REVIEW

File: conans/client/conf/__init__.py [hunk 1]
```diff
@@ -68,14 +70,26 @@
         threads: [None, posix, win32] #  Windows MinGW
         exception: [None, dwarf2, sjlj, seh] # Windows MinGW
         cppstd: [None, 98, gnu98, 11, gnu11, 14, gnu14, 17, gnu17, 20, gnu20]
-    Visual Studio:
+        intel:
+            None:
+            compatible:
+                version: [None, "11", "12", "13", "14", "15", "16", "17", "18", "19"]
```
Comment (danimtb) [inline conans/client/conf/__init__.py]: The additional `None` value here is to be able to do things like `conan install . --profile visual -s compiler.intel=compatible` without specifying the intel compiler version

Comment (SSE4) [reply conans/client/conf/__init__.py]: is it possible to use an anchor to avoid repetition of intel versions 4 times?

Comment (danimtb) [reply conans/client/conf/__init__.py]: In this case, we have the additional value of None, so I don't know if using anchoring is possible. Anyway, I don't like the settings structure either, but this is just a proposal.

**Comment by ohanar** [comment]
I personally prefer the approach in #5626.

**Comment by lasote** [comment]
Follow here: https://github.com/conan-io/conan/pull/6052

**Pull request closed without merging**